ethereum_history_api/circuits/lib/src/rlp/README.md

```diff
@@ -0,0 +1,100 @@
+# RLP decoding
+
+This is a [Noir](https://noir-lang.org) library to decode [RLP](https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/)
+
+RLP encoding works with two types of items. A string & a list. Therefore this lib contains two main functions:
+
+- ```rust
+  pub fn decode_string<N>(data: Fragment) -> RlpFragment;
+  ```
+- ```rust
+  pub fn decode_list<N, NUM_FIELDS>(data: Fragment) -> RlpList<NUM_FIELDS>;
+  ```
+- There is also a more performant special case version of `decode_list`, that can be used when we know that our input consists exclusively of strings of length <= 55 bytes.
+  ```rust
+  pub fn decode_list_of_small_strings<N, NUM_FIELDS>(data: Fragment) -> RlpList<NUM_FIELDS>;
+  ```
+
+## Example
+
+```rust
+let rlp = [0x82, 0x10, 0x00]; // "0x1000"
+decode_string(rlp.into());
+// {offset: 1, length: 2, data_type: STRING}
+
+let rlp = [0xc2, 0x80, 0x10]; // ["", "0x10"]
+let decoded = decode_list(rlp.into());
+// [{offset: 2, length: 0, data_type: STRING}, {offset: 2, length: 1, data_type: STRING}]
+
+// Or using our helper assert methods:
```

Comment:
```suggestion
// Or using helper assert methods:
```

Reply:
Also I think these 3 lines worsen the readability as it is not clear how `assert_eq_empty_string` and `assert_eq_u64` work

Reply:
But we wanted to show these functions.